sigstore/_internal/sct.py

```diff
@@ -149,6 +154,34 @@ def _get_issuer_cert(chain: List[Certificate]) -> Certificate:
     return issuer
 
 
+class UnexpectedSctCountException(Exception):
+    """
+    Number of percerts scts is wrong
+    """
+
+    pass
+
+
+def _get_sct_from_certificate(
+    certificate: Certificate,
+) -> Any:
```

Comment:
Hmm, can we refine the type here? This should return a `cryptography.x509.SignedCertificateTimestamp`, not `Any`: <https://cryptography.io/en/latest/x509/certificate-transparency/#cryptography.x509.certificate_transparency.SignedCertificateTimestamp>

Reply:
Actually I did that before, but I had problem with the linter, because [here](https://github.com/sigstore/sigstore-python/blob/8ea2a7c2c3664f932122a53daa67a09411beac49/sigstore/_internal/fulcio/client.py#L278C13-L278C16) the variable sct can be a `SignedCertificateTimestamp` but [here](https://github.com/sigstore/sigstore-python/blob/8ea2a7c2c3664f932122a53daa67a09411beac49/sigstore/_internal/fulcio/client.py#L301) it can be a `DetachedFulcioSCT`, and the linter doesn't understand the inheritance.

Reply:
I did a workaround for this by returning the `PrecertificateSignedCertificateTimestamps` and getting the `SignedCertificateTimestamp` on the caller flow instead.